aten/src/ATen/native/Math.h

```diff
@@ -191,6 +191,20 @@ C10_HOST_DEVICE static inline scalar_t zeta(scalar_t x, scalar_t q) __ubsan_igno
     }
   }
 
+  /* Asymptotic expansion
+   * https://dlmf.nist.gov/25.11#E43
+   */
+  if (q > acc_t{1e8}) {
+    return (one / (x - one) + one / (acc_t{2} * q)) * ::pow(q, one - x);
+  }
+
+  /* Euler-Maclaurin summation formula */
+
+  /* Permit negative q but continue sum until n+q > +9 .
+   * This case should be handled by a reflection formula.
+   * If q<0 and x is an integer, there is a relation to
+   * the polyGamma function.
+   */
   s = ::pow(q, -x);
```

Comment:
@kshitij12345 I found that reusing this `pow` had less of an impact on compilation times:
```cpp
  s = std::pow(q, -x);

  if (q > acc_t{1e8}) {
    return (q / (x - one) + acc_t{0.5}) * s;
  }
```

Reply:
Interesting. Thanks for looking into it! @peterbell10